nanoFramework.CoreLibrary/System/ArgumentException.cs

```diff
@@ -70,27 +75,68 @@ public ArgumentException(String message, String paramName)
         /// <value>
         /// A text string describing the details of the exception.
         /// </value>
-        public override String Message
+        public override string Message
         {
             get
             {
                 var s = base.Message;
-                if (!(_paramName == null || _paramName.Length == 0))
-                    return s + "\n" + "Invalid argument " + "'" + _paramName + "'";
+                if (!string.IsNullOrEmpty(_paramName))
+                {
+                    s += " " + string.Format(Arg_ParamName_Name, _paramName);
+                }
+
                 return s;
             }
         }
 
         /// <summary>
         /// Gets the name of the parameter that causes this exception.
         /// </summary>
-        /// <value>
-        /// The parameter name.
-        /// </value>
-        public virtual String ParamName
+        /// <value>The parameter name.</value>
+        public virtual string? ParamName => _paramName;
+
+        /// <summary>Throws an exception if <paramref name="argument"/> is null or empty.</summary>
+        /// <param name="argument">The string argument to validate as non-null and non-empty.</param>
+        /// <param name="paramName">The name of the parameter with which <paramref name="argument"/> corresponds.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="argument"/> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="argument"/> is empty.</exception>
+        public static void ThrowIfNullOrEmpty([NotNull] string? argument, [CallerArgumentExpression(nameof(argument))] string? paramName = null)
+        {
+            if (string.IsNullOrEmpty(argument))
+            {
+                ThrowNullOrEmptyException(argument, paramName);
+            }
+        }
+
+        /* TODO: Implement when string.IsNullOrWhiteSpace exists
+        /// <summary>Throws an exception if <paramref name="argument"/> is null, empty, or consists only of white-space characters.</summary>
+        /// <param name="argument">The string argument to validate.</param>
+        /// <param name="paramName">The name of the parameter with which <paramref name="argument"/> corresponds.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="argument"/> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="argument"/> is empty or consists only of white-space characters.</exception>
+        public static void ThrowIfNullOrWhiteSpace([NotNull] string? argument, [CallerArgumentExpression(nameof(argument))] string? paramName = null)
         {
-            get { return _paramName; }
+            if (string.IsNullOrWhiteSpace(argument))
+            {
+                ThrowNullOrWhiteSpaceException(argument, paramName);
+            }
         }
+        */
 
+        [DoesNotReturn]
+        private static void ThrowNullOrEmptyException(string? argument, string? paramName)
```

Comment:
If the extra method is an issue I could roll this into the main call. Was mainly just copying the code from .NET full for maintainability

Reply:
Understood. Moving it there won't make difference, just a couple of bytes with the method entry in the list.